File: server/router/modules/storage.go
package modules

import (
	"github.com/gin-gonic/gin"

	v1 "server/api/v1"
	"server/router/registry"
)

func init() {
	RegisterModule(&StorageModule{})
}

type StorageModule struct{}

func (m *StorageModule) Name() string {
	return "存储管理"
}

func (m *StorageModule) RegisterPublicRoutes(rg *gin.RouterGroup) {
	// 无公开路由
}

func (m *StorageModule) RegisterPrivateRoutes(rg *gin.RouterGroup) {
	R(rg, "GET", "/storages", m.Name(), "存储配置列表", v1.Storage.GetStorageList, registry.WithAuth())
	R(rg, "GET", "/storages/:id", m.Name(), "存储配置详情", v1.Storage.GetStorage, registry.WithAuth())
	R(rg, "POST", "/storages", m.Name(), "创建存储配置", v1.Storage.CreateStorage, registry.WithAuth())
	R(rg, "PUT", "/storages/:id", m.Name(), "更新存储配置", v1.Storage.UpdateStorage, registry.WithAuth())
	R(rg, "DELETE", "/storages/:id", m.Name(), "删除存储配置", v1.Storage.DeleteStorage, registry.WithAuth())
	R(rg, "PUT", "/storages/:id/default", m.Name(), "设置默认存储", v1.Storage.SetDefaultStorage, registry.WithAuth())
	R(rg, "POST", "/storages/test", m.Name(), "测试存储配置", v1.Storage.TestStorage, registry.WithAuth())
}

File: web/src/api/storage.ts
import request from '@/utils/request'
import type { Storage } from '@/types/storage'

// 获取存储配置列表
export function getStorageList() {
  return request.get<Storage[]>('/storages')
}

// 获取存储配置详情
export function getStorage(id: number) {
  return request.get<Storage>(`/storages/${id}`)
}

// 创建存储配置
export function createStorage(data: Partial<Storage>) {
  return request.post<Storage>('/storages', data)
}

// 更新存储配置
export function updateStorage(id: number, data: Partial<Storage>) {
  return request.put(`/storages/${id}`, data)
}

// 删除存储配置
export function deleteStorage(id: number) {
  return request.delete(`/storages/${id}`)
}

// 设置默认存储
export function setDefaultStorage(id: number) {
  return request.put(`/storages/${id}/default`)
}

// 测试存储配置
export function testStorage(data: Partial<Storage>) {
  return request.post('/storages/test', data)
}
